<launch>
    <arg name="usb_port" default="/dev/ttyUSB0"/>
    <arg name="baud_rate" default="57600"/>

    <param name="dynamixel_info"
        value="$(find test_dynamixel)/config/dynamixel_info_vel.yaml"/>
    <node
        name="dynamixel"
        pkg="dynamixel_workbench_controllers"
        type="dynamixel_workbench_controllers"
        required="true" output="screen"
        args="$(arg usb_port) $(arg baud_rate)">
        <param name="use_moveit" value="false"/>
        <param name="use_joint_states_topic" value="true"/>
        <param name="use_cmd_vel_topic" value="true"/>
        <rosparam>
            publish_period: 0.010
            dxl_read_period: 0.010
            dxl_write_period: 0.010
        </rosparam>
    </node>

    <node
        name="rqt_gui"
        pkg="rqt_gui"
        type="rqt_gui"
        args="--perspective-file $(find test_dynamixel)/config/service_call.perspective"
        />
</launch>
